### DISCUSSION TITLE
Orbits.jl

### DISCUSSION THREAD

**Discussion by mileslucas** [JuliaAstro/Transits.jl#31] [Ideas]
Is it too early to start discussing moving the orbit code to a new repo @icweaver ? Also perhaps there's some collaboration we can make with @sefffal directorbits code.

**Comment by sefffal** [comment]
@mileslucas I would be quite interested in joining efforts! I was thinking of registering that package soon.
Perhaps we can make a general PlanetOrbits (or KeplerianOrbits?) package.

From what I see, your `keplerian.jl` file is pretty much complete and has lots of nice helpers. I also really like that it supports units.
By contrast, [DirectOrbits.jl](https://github.com/sefffal/DirectOrbits.jl) is not as clean. Most of the computations are shoved into `kep2cart` and all the units are just be convention.

For me, one requirement is absolutely zero allocations and autodiff support. Right now ForwardDiff is supported but not Zygote, but I would like to fix that.

**Reply by icweaver** [reply]
I think that's a wonderful idea! I remember Miles and I had briefly discussed at one point about splitting out some functionality, so I think something like a `[Planet]Orbits.jl` package would be really cool. I don't any experience with direct imaging, but I really like how `kep2cart` handles radial velocities; it's something that's definitely been missing from our package compared to others like `exoplanet`.

Would people be up for discussing ideas during a call early next year?

**Reply by sefffal** [reply]
Great! Yes, absolutely I would.

RV is still a work in progress in our [DirectDections.jl](https://sefffal.github.io/DirectDetections.jl/dev/) package, but we're hoping to flesh it out this winter.

**Reply by sefffal** [reply]
There is also some relevant stuff in AstroBase.jl: 
https://github.com/JuliaAstro/AstroBase.jl/tree/master/src/Coords
